snuba/web/db_query.py

```diff
@@ -317,6 +318,25 @@ def _record_rate_limit_metrics(
         )
 
 
+def _apply_thread_quota_to_clickhouse_query_settings(
+    query_settings: QuerySettings,
+    clickhouse_query_settings: MutableMapping[str, Any],
+    project_rate_limit_stats: Optional[RateLimitStats],
+) -> None:
```

Comment:
Not a big deal because this method is private, but I think it would be nice (as a convention) to return a new `QuerySettings` with `max_threads` set and not modify the input.